java/src/main/native/src/row_conversion.cu

```diff
@@ -304,20 +334,600 @@ copy_from_fixed_width_columns(const cudf::size_type start_row, const cudf::size_
   }
 }
 
+#if !defined(__CUDA_ARCH__) || __CUDA_ARCH__ >= 700
+
+/**
+ * @brief The GPU blocks work on one or more block_info structs of data.
+ *        This structure defined the workspace for the block.
+ *
+ */
+struct block_info {
+  int start_col;
+  int start_row;
+  int end_col;
+  int end_row;
+  int batch_number;
+
+  constexpr size_type get_shared_row_size(size_type const *const col_offsets,
```

Comment:
I am having trouble understanding how `get_shared_row_size()` could be a `constexpr` function. `col_offsets` and `col_sizes` are pointers, i.e. runtime values.

Reply:
The mechanics of this code are eluding me. Since this is called from the `copy_to_rows()` kernel, I would have thought this would be a `__device__` method. Is the `constexpr` tag used here as an alternative to `__device__`?

Reply:
I believe the original code was more hard-coded than this and it morphed without the structure changing. I have adjusted these to be device functions and dropped the constexpr.